### ISSUE TITLE
ansible-lint fails if not running in a directory that is a .git repo

### ISSUE THREAD

**Issue by geerlingguy** [ansible/ansible-lint#763]
##### Summary

It would seem that in https://github.com/ansible/ansible-lint/pull/650, functionality was added that allowed `ansible-lint` to discover YAML files to lint if a git repository was not available.

However, a test playbook I created today didn't work, when I was in a directory that was not part of a git repo.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible 2.9.7
ansible-lint 4.2.0
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

macOS Catalina

##### STEPS TO REPRODUCE

Create a new directory outside of a git repo:

    mkdir playbook && cd playbook

Create a playbook `main.yml` inside with the contents:

```yaml
- hosts: localhost
  gather_facts: false
  connection: local

  tasks:
    - name: Register the output of the 'uptime' command.
      shell: test
      register: system_uptime
```

Run `ansible-lint` in the same directory as the playbook.

##### Desired Behaviour

ansible-lint detects yaml files without a .git repo present in the directory heirarchy.

##### Actual Behaviour

```
$ ansible-lint   
fatal: not a git repository (or any of the parent directories): .git
Traceback (most recent call last):
  File "/usr/local/bin/ansible-lint", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/ansiblelint/__main__.py", line 153, in main
    args = get_playbooks_and_roles(options=options)
  File "/usr/local/lib/python3.7/site-packages/ansiblelint/utils.py", line 774, in get_playbooks_and_roles
    universal_newlines=True).split()))
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 411, in check_output
    **kwargs).stdout
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 512, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['git', 'ls-files', '*.yaml', '*.yml']' returned non-zero exit status 128.
```

##### Workarounds

For now, the best workaround is to init a git repo in the same directory, and add any playbooks you want `ansible-lint` to discover to it:

    git init .
    git add -A
    git commit -m "Added stuff so ansible-lint can pick it up."

And then run it again.

Or, if you don't want to create a repo, you could pass any relevant files directly to the command as arguments to bypass the discovery mechanism, e.g.:

    ansible-lint main.yml

**Comment by geerlingguy** [comment]
Hmm... I just noticed that the commit from the PR I mentioned earlier (https://github.com/ansible/ansible-lint/commit/49ca5b5374dcd972b55903c9bfa10332cfc3eb37) was merged after the 4.2.0 release was tagged. It seems to be in the `4.3.0a0` release, which I'll try momentarily.

**Comment by geerlingguy** [comment]
Well, after installing `4.3.0a0`, I didn't get a fatal error anymore, but I didn't get success. Instead, it just prints out the `--help` contents. If I'm reading the PR correctly, it _should_ be falling back to trying to discover `yml`/`yaml` files via Python instead of git, but it seems to not be doing that.

**Comment by webknjaz** [comment]
This was also broken after and got hotfixes after the last alpha tag. Try master too.

**Comment by webknjaz** [comment]
I bet @ssbarnea can explain better.

**Comment by ssbarnea** [comment]
The auto-detect was bit buggy but went much worse when it vanished during a refactoring change, so you are better of avoiding 4.3 and even 4.2 if you are not passing playbooks directly. 

Is very easy to spot the fact that `get_yaml_files` is no longer called from anywhere. Doing a git blame and adding it back would clearly help. I was planning to do it, but other issues preventing me from fixing it.

In that this issue was the reason why I raised https://github.com/ansible/ansible-lint/issues/758 as this would have prevented such a serious regression from happening.
